components/public-api-server/pkg/apiv1/workspace.go

```diff
@@ -221,6 +206,38 @@ func (s *WorkspaceService) UpdatePort(ctx context.Context, req *connect.Request[
 	), nil
 }
 
+func (s *WorkspaceService) StartWorkspace(ctx context.Context, req *connect.Request[v1.StartWorkspaceRequest]) (*connect.Response[v1.StartWorkspaceResponse], error) {
```

Comment:
Do we need to explicitly handle the case (the response) when the WS is already running?

Reply:
Don't think so :thinking: , this is already handled by private api, and then public api will return workspace object data as expected (cached internally by private api if no state have not changed)

Reply:
It is handled [here](https://github.com/gitpod-io/gitpod/blob/se/reduce-gc-load/components/server/src/workspace/workspace-starter.ts#L302-L309).